plugins/org.eclipse.embedcdt.managedbuild.cross.riscv.core/src/org/eclipse/embedcdt/managedbuild/cross/riscv/core/EnvironmentVariableSupplier.java

```diff
@@ -89,28 +92,21 @@ public static PathEnvironmentVariable create(IConfiguration configuration) {
 
 			Boolean preferXpacksBin = Option.getOptionBooleanValue(configuration, Option.OPTION_PREFER_XPACKS_BIN);
 
-			String path = "";
+			List<String> path = new ArrayList<>();
 			if (preferXpacksBin) {
-				IPath projectPath = project.getWorkspace().getRoot().getLocation().append(project.getFullPath());
-				IPath xpackBinPath = projectPath.append("xpacks").append(".bin");
-
-				path = xpackBinPath.toOSString();
+				String xpackBinPath = project.getFolder("xpacks").getFolder(".bin").getLocation().toOSString();
+				path.add(xpackBinPath);
```

Comment:
Does this produce the absolute path of the `xpacks/.bin` folder located inside the project?

Reply:
This fixed a bug that the code assumed that the project was located on disk in the workspace folder.

e.g. if I have a project on my computer in `E:\cdt\git\armproj` and my workspace is `E:\cdt\runtime-CDT` then the old code would add `E:\cdt\runtime-CDT\armproj\xpacks\.bin` to the `path` variable, when it should have added `E:\cdt\git\armproj\xpacks\.bin`

This was happening because of the order of resolution. The updated code gets the full internal path to the `bin` first (e.g. `project.getFolder("xpacks").getFolder(".bin")` == `F/armproj/xpacks/.bin`) and then gets the location on disk of that folder. The old code (`project.getWorkspace().getRoot().getLocation()`) got the location on disk of the workspace and then appended paths to it.

Let me know if you want me to split that out to a different PR.

Reply:
> Does this produce the absolute path of the xpacks/.bin folder located inside the project?

TL;DR - Yes! :)